src/main/ruby/core/symbol.rb

```diff
@@ -138,3 +138,11 @@ def [](index, other = undefined)
   # Use equal? for ===
   alias_method :===, :equal?
 end
+
+module Truffle::SymbolOperations
+  
+  def self.is_constant?(name)
+    name =~ /^[A-Z][[:alnum:]_]*$/
```

Comment:
This would pass for names with newlines in it. Use `\A`/`\z` to really match beginning/end-of-string.